nemo/collections/nlp/modules/common/text_generation_utils.py

```diff
@@ -553,7 +669,7 @@ def sample_sequence_batch(
                 # make sure it won't sample outside the vocab_size range
                 logits[:, tokenizer.vocab_size :] = -float('Inf')
```

Comment:
I have this guard to make sure it won't sample outside the `vocab_size`. I see you also do `Replace special soft prompt token ids with unk token ids` later. Isn't it better to make sure the model never samples those special soft prompt tokens?

Reply:
I added that replacement step to stop out of bounds indexing errors related to virtual prompt id values being higher than the number of logits predicted. The virtual prompt id values were sometimes greater than the size of the last dimension of `logits` and `output`. I only apply the replacement to the new token tensor to address soft prompt input tokens for items in the batch that haven't started generating new tokens yet even though others have.

Reply:
Yes, it does. Though they're still included in the list of tokens and long probs portion of the output. So someone using those will have to filter them out before calculating and metrics. But this is true even for padding or EOS tokens, so shouldn't be an issue.